### PR TITLE
Select forward ref LG-3681

### PR DESCRIPTION
**Pull request by TheSonOfThomp**
## ✍️ Proposed changes

Allows Select to accept a ref

🎟 _Jira ticket:_ https://jira.mongodb.org/browse/LG-3681

### For bug fixes, new features & breaking changes

- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [x] I have run `yarn changeset` and documented my changes

### CODE REVIEW

File: packages/select/src/Select/Select.tsx [hunk 1]
```diff
@@ -43,563 +52,576 @@ import { DropdownWidthBasis, SelectProps, Size, State } from './Select.types';
...
-  const onClose = useCallback(() => {
-    setOpen(false);
-    menuButtonRef.current!.focus();
-  }, [menuButtonRef]);
-
-  useEffect(() => {
-    if (!open) {
-      return;
-    }
+    const containerRef = useForwardedRef(fwdRef, null);
```
Comment (bruugey) [inline packages/select/src/Select/Select.tsx:108]: is this the right place to forward teh ref?

Comment (TheSonOfThomp) [reply packages/select/src/Select/Select.tsx:108]: technically yes, should be on the container. I'd _prefer_ it to be on the button element itself, but then there's no way to  reference the Label/Description

File: .changeset/loud-trainers-sparkle.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@leafygreen-ui/select': minor
```
Comment (bruugey) [inline .changeset/loud-trainers-sparkle.md:2]: major change. per react docs wrapping something in fwd ref should be a breaking change

Comment (TheSonOfThomp) [reply .changeset/loud-trainers-sparkle.md:2]: Really? I believe you but can you link the docs?

**Review by bruugey — APPROVED** [review]

**Pull request merged**